src/main/java/seedu/weme/model/util/MemeUtil.java

```diff
@@ -0,0 +1,60 @@
+package seedu.weme.model.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import seedu.weme.commons.util.FileUtil;
+import seedu.weme.model.meme.ImagePath;
+import seedu.weme.model.meme.Meme;
+
+/**
+ * A utility class for operations related to {@code Meme}.
+ */
+public class MemeUtil {
+
+    /**
+     * Copies the image of {@code toCopy} to {@code ImagePath}, and returns a new {@code Meme} whose {@code ImagePath}
+     * points to that location.
+     *
+     * @param toCopy       the {@code Meme} to copy
+     * @param memeLocation the meme image location
+     * @return a new {@code Meme} with the new {@code ImagePath}.
+     */
+    public static Meme copyMeme(Meme toCopy, Path memeLocation) throws IOException {
+        Path originalPath = toCopy.getFilePath().getFilePath();
+        Path newPath = getNewImagePath(originalPath, memeLocation);
+        FileUtil.copy(originalPath, newPath);
+        return new Meme(new ImagePath(newPath.toString()), toCopy.getDescription(), toCopy.getTags());
+    }
+
+    /**
+     * Similar to {@link MemeUtil#copyMeme(Meme, Path)} except this method deletes the copied image file afterwards.
+     *
+     * @param toCopy       the {@code Meme} to copy
+     * @param memeLocation the meme image location
+     * @return a new {@code Meme} with the new {@code ImagePath}.
+     */
+    public static Meme generateCopiedMeme(Meme toCopy, Path memeLocation) throws IOException {
```

Comment:
so this method is basically just to get the copied meme object. Can you rephrase the comment to reflect that? It sounds like it the file stays for a while when it is actually deleted immediately.